cfgov/v1/util/migrations.py

```diff
@@ -57,3 +61,73 @@ def get_or_create_page(apps, page_cls_app, page_cls_name, title, slug,
     parent_page.save()
 
     return page
+
+
+def get_stream_data(page_or_revision, field_name):
+    """ Get the stream field data for a given field name on a page or a
+    revision """
+    try:
+        # If page_or_revision is a page, this will work
+        field = getattr(page_or_revision, field_name)
+        stream_data = field.stream_data
+    except AttributeError:
+        # Otherwise it will raise an Attribute error and we can assume
+        # page_or_revision is a revision
+        revision_content = json.loads(page_or_revision.content_json)
+        field = revision_content[field_name]
+        stream_data = json.loads(field)
+    return stream_data
+
+
+def set_stream_data(page_or_revision, field_name, stream_data):
+    """ Set the stream field data for a given field name on a page or a
+    revision """
+    try:
+        # If page_or_revision is a page, this will work
+        field = getattr(page_or_revision, field_name)
+        stream_block = field.stream_block
+        stream_value = StreamValue(stream_block, stream_data, is_lazy=True)
+        setattr(page_or_revision, field_name, stream_value)
+        page_or_revision.save()
+    except AttributeError:
+        # Otherwise it will raise an Attribute error and we can assume
+        # page_or_revision is a revision
+        revision_content = json.loads(page_or_revision.content_json)
+        revision_content[field_name] = json.dumps(stream_data)
+        page_or_revision.content_json = json.dumps(revision_content)
+        page_or_revision.save()
+
+
+def migrate_stream_field(page_or_revision, field_name, field_type, mapper):
+    """ Migrate a block of the type within a StreamField of the name belonging
+    to the page or revision using the mapper function """
+    old_stream_data = get_stream_data(page_or_revision, field_name)
+    new_stream_data = []
+
+    migrated = False
+    for field in old_stream_data:
+        if field_type == field['type']:
+            field['value'] = mapper(page_or_revision, field['value'])
+            migrated = True
+
+        new_stream_data.append(field)
+
+    if migrated:
+        set_stream_data(page_or_revision, field_name, new_stream_data)
+
+
+@transaction.atomic
+def migrate_page_types_and_fields(apps, page_types_and_fields, mapper):
+    """ Migrate the fields of a wagtail page type using the given mapper
+        function. page_types_and_fields should be a list of 3-tuples
+        providing ('PageType', 'field_name', 'block type'). """
+    for page_type, field_name, field_type in page_types_and_fields:
+        page_model = apps.get_model('v1', page_type)
```

Comment:
Yup, I meant to do that, actually. 

I was thinking this would be a nice thing to have live in a Wagtail contrib module in our fork of Wagtail... whenever we do that.